core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1493,9 +1493,7 @@ public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
         auditContext.setAllowed(false);
         throw e;
       }
-      if (!options.isAlluxioOnly()) {
-        mMountTable.checkUnderWritableMountPoint(path);
-      }
+      mMountTable.checkUnderWritableMountPoint(path);
```

Comment:
this means we cannot remove a file only from Alluxio, after it has been loaded?

Reply:
nope. you will still be able to delete from alluxio only. for some reason, an alluxio only delete was not checking the readOnly flag before. do you know why?

Reply:
it doesn't have to exist only in alluxio. rm --alluxioOnly is also applicable if file is in alluxio and ufs

Reply:
I see. yeah we should fix that. Good thing is that user can recover from this case by running `alluxio fs ls -f`.

Reply:
I'm confused. I thought the previous code allowed an alluxio-only delete (file exists in alluxio and ufs) to work, even if it was a read-only mount. In the new code, it will disallow even an alluxio-only delete, right? Is that what we want?

Reply:
yes. read-only mount and alluxio-only are 2 independent concepts. previously, we disallowed createFile, createDirectory, rename under a read-only mount (regardless of PERSISTED or not). delete was a special case, as you pointed out, an alluxio-only delete under a read-only mount point was allowed. I think we should be consistent and disallow all write operations (including delete and setAttribute) regardless of alluxio-only.

Reply:
I see, that makes sense. Thanks!